### ISSUE TITLE
Standalone webpage for browsing presets

### ISSUE THREAD

**Issue by 1ec5** [openstreetmap/id-tagging-schema#646]
We should create a standalone webpage for browsing id-tagging-schema presets that’s easy to use without knowledge of JSON or this repository’s structure.

## Background

In tagging or onboarding discussions, I often find myself using [the standalone iD preview on Netlify](https://ideditor.netlify.app/) or this GitHub repository’s search box to translate a real-world term into a tagging suggestion or vice versa. This workflow requires some insider knowledge about how the repository is structured or how to use iD’s raw tag editor.

People often refer newcomers to the OSM Wiki’s “[Map features](https://wiki.openstreetmap.org/wiki/Map_features)” page because it offers a high-level overview of all the available tags on one page, a better starting point than using the OSM Wiki’s search engine. Ideally, an iD user should be able to access all the tagging documentation they need from within iD, but sometimes people need to refer to tags in other contexts. Users of other editors may gravitate toward documentation that lives outside the editor.

Unfortunately, the “Map features” page is reaching the limit of MediaWiki’s capabilities, because it has to transclude content and links from a very large number of independent pages. This problem is essentially the same whether the page is based on transcluding disparate content pages or data items, because MediaWiki’s core content model is a document-oriented database. Taginfo can make popular tags available, but it only traffics in raw tags, and the documentation it extracts from the wiki has large gaps, especially outside of a handful of languages, because it ignores the translated descriptions in data items. Besides, the wiki doesn’t maintain names of features, only descriptions.

“Map features” is the product of a good deal of curation. In principle, one could create a listing similar to “Map features” using a simple SPARQL query in Sophox, but even if Sophox/sophox#31 were fixed, the listing would be as untidy as the wiki itself. This repository is also curating tags, but at a higher level of abstraction. The presets in this repository would be more educational to mappers than a table that only lists one tag for each feature, even when a tag may only makes sense in combination with other tags.

## Proposal

A simple standalone webpage would list each preset in this repository, organized by category, listing any aliases and the tags applied. Selecting a preset could show a brief description and image based on either taginfo or the linked data item, like in iD, with a link out to the wiki. There could be an option to set the language and region of the presented presets.

Publishing this webpage probably doesn’t require anything more complex than GitHub Pages. id-tagging-schema’s built package is already distributed over the jsDelivr CDN, so it could be just a static webpage that loads the latest JSON files from there. A static webpage can be deployed anywhere. It could even be implemented as a gadget or user script on the wiki itself.

In contrast to osmlab/osm-planning#27, I don’t envision this webpage taking over the wiki in general, but it would serve as an alternative to the “Map features” page that’s increasingly broken and has never served as many language communities as this project.

**Comment by mnalis** [comment]
I'd like that. Searching the wiki directly is a chore (and `Map Features` especially has [problems](https://community.openstreetmap.org/t/data-item-not-found-check-your-spelling-for-amenity-mailroom/5618/2?u=matija_nalis)), and [Tagfinder](https://wiki.openstreetmap.org/wiki/TagFinder) is also having problems finding things recently.

**Comment by matkoniecz** [comment]
Should it live as part of this repository and be maintained here? Or would it make more sense to exist as more separate project, potentially maintained by someone else?

**Comment by RevitBlessed** [comment]
Hi! I am the author of draft solution #2119 . I’m sharing here the link to the prototype and the repository:


https://osm-id-tagging.netlify.app/

https://github.com/RevitBlessed/OSM-iD-Tagging-Schema

I would really appreciate any feedback on the current state — what should be improved, fixed, or reconsidered.
I understand that I might not fully grasp all the details of how presets work, since the system is quite complex.

I also have a few specific questions:


### 1. Preset vs Template

In cases like `education/music_school`, there is both a preset and a corresponding template `{education/music_school}`.

What would be the correct approach:

* to display them separately (as I currently do),
* or to combine them into a single resolved result (preset + template)?

Also, is it correct that the final tags should come from the template (e.g. `amenity=music_school`), rather than being inferred from the preset name?


### 2. Field inheritance and UI structure

In the prototype, I currently show the origin of each field (for example: from `contact`, `poi`, or a parent preset), so it’s clear how the final set of fields is formed.

From a UX perspective, what would be better:

* to show only the final resolved preset (without details),
* or to explicitly show templates / parent presets and the origin of each field?


### 3. Regional field variants

Some fields (for example `highway/footway/crossing/unmarked`) have region-specific variants.

What would be the better approach:

* is it enough to indicate that a field has regional differences,
* or is it important to explicitly show how these variants differ (e.g. different values per country)?

How important is it to expose these differences from a UX perspective?


### 4. Possible improvements

I am also considering extending the prototype with:

* semantic search (e.g. “food” → restaurant, cafe, etc.)
* descriptions and images similar to Map Features (where available)
* possibly using the tool to help with the icon issue discussed in #2208 

Do these ideas align well with the expected scope?

Thanks!

**Comment by tordans** [comment]
I have been experimenting this over the last couple of days.
You can see the results at 
* Regular UI: https://osmberlin.github.io/tagging-schema-browser/?locale=de
* Github: https://github.com/osmberlin/tagging-schema-browser
* PR UI: https://osmberlin.github.io/tagging-schema-browser/comparison?dataUrl=https://pr-2382--ideditor-presets-preview.netlify.app/dist/&locale=de&q=&page=1&per_page=24&sort=name_asc&primaryTagKey=[]&geometry=[]&iconPrefix=[]&iconName=[]&fieldIds=[]&categoryNames=[]&hasIcon=[]

I hope this exploration helps us to find the right use cases for this and based on that adjust the UI.

**My focus was:**
1. A UI that helps **explore the icon libraries**, unused icons, used icons and how often / by which preset etc. => This can help with finding icons and cleaning up double usage.

2. A UI that helps **comparing translations**, especially search relevant strings. Tobias Z wrote about this in a forum comment: We can improve the usability by a lot by improving the findability, especially with local strings. This UI helps a bit to compare coal translations (change lang at the top right) with the core translations. Ideally we build a connection to the translations system but it looks like we need a server (not a SPA) for that – lets explore ideas more…

3. The main presets can be **listed in a compare like UI**. That helps to spot things like the `bench` field on one playground preset https://osmberlin.github.io/tagging-schema-browser/?locale=de&categoryNames=%5B%22Playground+Equipment%22%5D which I did not expect (might be right; just an example…). Not sure about this page, though, we can refine the use cases here for sure.

4. The page **compares a released version** with a branch deployment. Use the about page to paste a deploy preview and the UI will show the diff to the released version. I image us adding this to our preview message in PRs so we can spot the diff of the PR against a reference. Playing around with this showed, that the reference to the released version is not ideal; we need something like a continuously updated "internal" or "prerelease" version ideally, so we only get the diff to the current main?

---

Status of this code: To me this is a prime candidate for LLM dev experimentation. It is very UI focussed. Has a view places that are complex but overall its more just displaying data. Nothing really happens if this breaks… — So if you have input, ideally write it like an LLM input (goal; current UI & where => new UI & where). Or create PRs in this spirit. — Once we are happy with the UX, I will do a code review to check the code is good enough.

CC @matkoniecz @1ec5 @tyrasd

**Comment by matkoniecz** [comment]
coal translations?

**Comment by danieldegroot2** [comment]
Ref [NLnet NGI0 Commons Fund - OpenStreetMap iD tagging schema overhaul](https://nlnet.nl/project/OSM-iD-tagging/)
<sub>( [description](https://nlnet.nl/news/2026/20260616-67-new-projects.html#:~:text=OpenStreetMap%20iD%20tagging%20schema%20overhaul) ); long description not available as of writing.</sub>

Ref [NLnet NGI0 Commons Fund - Comparison + readability of OSM iD presets](https://nlnet.nl/project/OSM-iD-presets/)
<sub>which is mentioned but not listed in above page</sub>

**Comment by matkoniecz** [comment]
@danieldegroot2 Yeah, I will either tell them that it become obsolete or retarget it to something else that is also useful. Process was going since 2025 and not even late 2025.

**Comment by matkoniecz** [comment]
> I hope this exploration helps us to find the right use cases for this and based on that adjust the UI.

do you want feedback on UI details?

if yes then my first comment would be that way of getting back to code would be nice, especially when file is mentioned - also linking it would be great

<img width="1076" height="942" alt="Image" src="https://github.com/user-attachments/assets/4b42f526-4c10-4005-8431-e14e24617ea9" />

**Comment by tordans** [comment]
> > I hope this exploration helps us to find the right use cases for this and based on that adjust the UI.
> 
> do you want feedback on UI details?
> if yes then my first comment would be that way of getting back to code would be nice, especially when file is mentioned - also linking it would be great

I created https://github.com/osmberlin/tagging-schema-browser/issues/4 for this. Feel free do create issues there with modification ideas and use case ideas. Will try to explore those quickly…
